imagelab_react/src/App.js

```diff
@@ -2,30 +2,22 @@ import './App.css';
 import './imagelab-block'
 import React, { useState, useEffect } from "react";
 import { BlocklyWorkspace } from "react-blockly";
+import Blockly from "blockly";
 import { MY_TOOLBOX } from './toolboxConfiguration';
 import { Navbar, Alignment, Button, Card, Elevation } from '@blueprintjs/core';
 import './CustomBlockly.css';
-import Blockly from "blockly";
 import ImageViewer from './components/ImageViewer';
 import { workspaceConfiguration } from './workspaceConfig';
+import { redo, reset, run, undo } from './utils/main';
 
 function App() {
   const [xml, setXml] = useState("");
   const [imageUrl, setImageUrl] = useState(null);
+  const [processedImage, setProcessedImage] = useState(null);
 
-  const handleUndo = () => {
-    Blockly.mainWorkspace.undo(false);
-  }
-
-  const handleRedo = () => {
-    Blockly.mainWorkspace.undo(true);
-  }
-
-  const handleReset = () => {
-    const shouldReset = window.confirm('Are you sure you want to reset the workspace?');
-    if (shouldReset) {
-      Blockly.mainWorkspace.clear();
-    }
+  const start = () => {
+    const topBlock = Blockly.getMainWorkspace().getTopBlocks()[0];
+    run(topBlock);
```

Comment:
what happened to the other two functions? @abdullahranginwala handleRedo and handleReset